src/middleware/TxBuilder/LedgerTxBuilder.ts

```diff
@@ -12,28 +12,27 @@ import TxBuilder from './TxBuilder';
 export default class LedgerTxBuilder extends TxBuilder {
   private tx!: LedgerTx;
 
-  buildTx(): Promise<LedgerTx> {
+  buildTx(normalizedTx: NormalizedTx, accountType: string):
+    Promise<LedgerTx> {
     return new Promise<LedgerTx>((resolve, reject) => {
       const { coin } = this;
-      if (this.normalizedTx) {
-        this.getLedgerTxData(this.normalizedTx)
-          .then(({ inputs, associatedKeysets, outputScriptHex }) => {
-            const tx: LedgerTx = {
-              outputs: this.normalizedTx.outputs,
-              outputScriptHex,
-              changePath: this.changeAddress,
-              coin,
-              inputs,
-              associatedKeysets,
-              accountType: this.accountType,
-            };
-            this.tx = tx;
-            resolve(tx);
-          })
-          .catch(reject);
-      } else {
-        reject(new Error('There is no Normalized transaction created'));
-      }
+      const changeAddress = normalizedTx.outputs[2].address
+        ? normalizedTx.outputs[2].address : '';
+      this.getLedgerTxData(normalizedTx)
+        .then(({ inputs, associatedKeysets, outputScriptHex }) => {
+          const tx: LedgerTx = {
+            outputs: normalizedTx.outputs,
+            outputScriptHex,
+            changePath: changeAddress,
```

Comment:
have you tried this when you don't have change?

Reply:
tested and works